spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MklInt8Convertible.scala

```diff
@@ -0,0 +1,427 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+
+import scala.collection.mutable.ArrayBuffer
+import scala.reflect.ClassTag
+
+/**
+ * Trait which provides MKL-DNN functionality to convert FP32 model to INT8 model
+ */
+trait MklInt8Convertible {
+  // input dimension mask
+  protected var inputDimMask: Int = 0
+  // output dimension mask
+  protected var outputDimMask: Int = 0
+  // weight dimension mask
+  protected var weightDimMask: Int = 0
+  // input activation scales
+  private[nn] var inputScalesBuffer: ArrayBuffer[Array[Float]] = ArrayBuffer.empty[Array[Float]]
+  // output scales
+  private[nn] var outputScalesBuffer: ArrayBuffer[Array[Float]] = ArrayBuffer.empty[Array[Float]]
+  // weight scales
+  private[nn] var weightScalesBuffer: ArrayBuffer[Array[Float]] = ArrayBuffer.empty[Array[Float]]
+
+
+  /**
+   * Calculate the required scales for converting int8 modules
+   * Currently there are four type of modules should be supported:
+   * 1) Linear: calculate scales for input, output and weight
+   * 2) Spatial Convolution: calculate scales for input, output and weight
+   * 3) Sequential: calculate scales for input, output as well as the scales of submodules
+   * 4) ConcatTable: calculate scales for input, output as well as the scales of submodules
+   * @param inActivity
+   */
+  def calcScales(inputActvt: Activity): Unit = {
```

Comment:
make it invisible out of BigDL, and do a module type check for only supporting Float